pkg/oci/spec_opts.go

```diff
@@ -190,41 +190,30 @@ func WithEnv(environmentVariables []string) SpecOpts {
 // replaceOrAppendEnvValues returns the defaults with the overrides either
 // replaced by env key or appended to the list
 func replaceOrAppendEnvValues(defaults, overrides []string) []string {
-	cache := make(map[string]int, len(defaults))
-	results := make([]string, 0, len(defaults))
-	for i, e := range defaults {
-		k, _, _ := strings.Cut(e, "=")
-		results = append(results, e)
-		cache[k] = i
-	}
-
-	for _, value := range overrides {
-		// Values w/o = means they want this env to be removed/unset.
-		k, _, ok := strings.Cut(value, "=")
-		if !ok {
-			if i, exists := cache[k]; exists {
-				results[i] = "" // Used to indicate it should be removed
-			}
-			continue
-		}
+	cache := make(map[string]int, len(defaults)+len(overrides))
+	merged := make([]string, 0, len(defaults)+len(overrides))
 
-		// Just do a normal set/update
-		if i, exists := cache[k]; exists {
-			results[i] = value
-		} else {
-			results = append(results, value)
+	for n, e := range append(defaults, overrides...) {
+		if !strings.Contains(e, "=") {
```

Comment:
Curious; could we continue using `strings.Cut` here? (also taking advantage of it returning whether it contains a `=` ?

Reply:
Sounds good. `strings.Contains` and `strings.SplitN(` can be replaced.